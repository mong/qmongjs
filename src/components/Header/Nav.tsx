import React, { useState, CSSProperties } from "react";
import { NavLink } from "react-router-dom";

import style from "./nav.module.css";

type HeaderProps = {};

const Nav: React.FC<HeaderProps> = () => {
  const [menuToggle, updateMenuToggle] = useState(false);
  const styleTop: CSSProperties = menuToggle ? { transform: "scale(0)" } : {};
  const styleMiddle: CSSProperties = menuToggle
    ? { transform: "translate(0px, 0px) rotate(45deg)  scale(1.2)" }
    : {};
  const styleBottom: CSSProperties = menuToggle
    ? { transform: "translate(0px, -7px) rotate(-45deg)  scale(1.2)" }
    : {};
  const styleBtn: CSSProperties = menuToggle ? { borderRadius: "100%" } : {};
  const styleNavLinks: CSSProperties = menuToggle ? { display: "flex" } : {};
  const handleClick = () => updateMenuToggle(!menuToggle);

  return (
    <div className={style.navBar}>
      <div className={style.navLinks} style={styleNavLinks}>
        <NavLink to="/kvalitetsregistre" activeStyle={{ fontWeight: 600 }}>
          Kvalitetsregistre
        </NavLink>
        <a href="https://helseatlas.no/">Helseatlas</a>
        <a href="https://helse-nord.no/skde/pasientstrommer">Pasientstr√∏mmer</a>
        <a href="https://helse-nord.no/skde">Om SKDE</a>
      </div>
      <button
        className={style.hamburgerbtn}
        onClick={() => handleClick()}
        style={styleBtn}
      >
        <span className={style.hamburgerWrapper}>
          <span className={style.lineTop} style={styleTop}></span>
          <span className={style.lineMiddle} style={styleMiddle}></span>
          <span className={style.lineBottom} style={styleBottom}></span>
        </span>
      </button>
    </div>
  );
};

export default Nav;
