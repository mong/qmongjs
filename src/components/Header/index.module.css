.header {
  display: flex;
  width: min(960px, calc(100% - 64px));
  align-items: center;
  justify-content: space-between;
  background-color: white;
  padding: 32px 0;
  margin: auto;
}
.header a {
  text-decoration: none;
  color: black;
  font-size: 14.4px;
  font-family: Arial;
}

.headerContainer {
  display: flex;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: flex-start;
  padding: 0.5rem 7%;
}

.logo {
  padding-right: 2rem;
  width: 100;
}

.logo_img {
  height: 39px;
  width: 96px;
}
