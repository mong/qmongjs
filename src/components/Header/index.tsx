import React from "react";
import style from "./index.module.css";
import skdeLogo from "../../img/SKDE_sort.png";

type HeaderProps = {};

const Header: React.FC<HeaderProps> = () => {
  return (
    <div style={{}}>
      <header className={style.header}>
        <div className={style.logo}>
          <a href={window.location.origin}>
            <img className={style.logo_img} src={skdeLogo} alt="SKDE logo" />
          </a>
        </div>
        <div>
          <a href={`${window.location.origin}/kontakt`}>Kontakt</a>
        </div>
      </header>
    </div>
  );
};

export default Header;
