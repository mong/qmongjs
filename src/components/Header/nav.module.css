.navBar {
  display: flex;
  justify-content: space-around;
  align-items: flex-start;
}

.navLinks {
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.navLinks a {
  text-decoration: none;
  color: #25407b;
  font-size: 1.2rem;
  padding: 10px;
}

.navLinks a:hover,
.navLinks a:focus {
  outline: unset;
  text-decoration: underline;
  color: #6da7df;
}

.hamburgerbtn {
  display: none;
  background: none;
  position: relative;
  height: 3rem;
  outline: unset;
  font-size: 1rem;
  width: 50px;
  border: 0;
}

.hamburgerbtn:hover,
.hamburgerbtn:focus {
  background: rgba(109, 167, 223, 0.3);
}

.hamburgerWrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  color: #25407b;
  text-decoration: none;
}

.lineTop,
.lineMiddle,
.lineBottom {
  width: 20px;
  height: 1px;
  background-color: #25407b;
  transition: transform 0.1s ease-out;
  margin: 3px 0;
}

@media All and (max-width: 768px) {
  .navBar {
    justify-content: flex-end;
  }
  .hamburgerbtn {
    display: flex;
    align-content: center;
    justify-content: center;
  }
  .navLinks {
    display: none;
    flex-direction: column;
  }
}
