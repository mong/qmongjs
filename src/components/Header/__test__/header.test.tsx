import React from "react";
import { BrowserRouter, Route } from "react-router-dom";
import { QueryParamProvider } from "use-query-params";
import { render } from "@testing-library/react";

import Header from "../";

it("main page header renders", async () => {
  const { container } = render(
    <BrowserRouter>
      <QueryParamProvider ReactRouterRoute={Route}>
        <Header />
      </QueryParamProvider>
    </BrowserRouter>
  );

  expect(container).toMatchSnapshot();
});
