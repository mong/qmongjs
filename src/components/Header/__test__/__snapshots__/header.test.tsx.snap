// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main page header renders 1`] = `
<div>
  <div>
    <header
      class="header"
    >
      <div
        class="logo"
      >
        <a
          href="http://localhost"
        >
          <img
            alt="SKDE logo"
            class="logo_img"
            src="SKDE_sort.png"
          />
        </a>
      </div>
      <div>
        <a
          href="http://localhost/kontakt"
        >
          Kontakt
        </a>
      </div>
    </header>
  </div>
</div>
`;
